roles/download/defaults/main.yml

```diff
@@ -31,11 +32,13 @@ flannel_version: "v0.8.0"
 flannel_cni_version: "v0.2.0"
 pod_infra_version: 3.0
 
-# Download URL's
+# Download URLs
 etcd_download_url: "https://storage.googleapis.com/kargo/{{etcd_version}}_etcd"
+kubeadm_download_url: "https://storage.googleapis.com/kubernetes-release/release/{{ kubeadm_version }}/bin/linux/amd64/kubeadm"
```

Comment:
@luxas Is there a containerized kubeadm release? Is it part of hyperkube?

Reply:
Not sure what you mean with containerized kubeadm. It is a CLI binary and has no deps.
It just writes down a couple of files and expects kubelet to be running.
You can run kubelet however you want, be it containerized, on-host or in an rkt pod or whatever

Reply:
And no, not part of hyperkube, by design.

Reply:
We have the option to run nearly every component in a container ( including helm ). I'd like to carry this model forward if possible with kubeadm. Especially in this case since we're just downloading an unpackaged binary, it makes it interesting when managing kubeadm versions and understanding what's actually installed.

Reply:
Ok. Neither kubectl or kubeadm is packaged in container images officially.
You can easily create your own image though.

Reply:
Yeah, that would be nice to have indeed. See https://github.com/kubernetes/kubernetes/issues/35041, didn't make it tho :)

Reply:
All , just wanted to understand why can't kubeadm support both systemd and containerized components. Since we are talking of kubeadm entity as a primary bootstrap mechanism the support  for components in systemd would be a catalyst for a larger community, I would love to weigh in on any ideas or any work someone can direct me to.